Source/SwiftLintBuiltInRules/Rules/Lint/InvalidSwiftLintCommandRule.swift

```diff
@@ -60,35 +62,54 @@ struct InvalidSwiftLintCommandRule: Rule, SourceKitFreeRule {
 
     private func styleViolation(for command: Command, in file: SwiftLintFile, reason: String) -> StyleViolation {
         let character = command.startingCharacterPosition(in: file)
+        let characterOffset = character.flatMap {
```

Comment:
So I think we could make the `characterOffset` retrieval a function on the `Command` extension, so this reads more nicely here, but having to do all of these line calculations feels a bit off in general.

The reason we have to do this calculations is because `Command`'s `character` property records the character beyond the end of the command. If that instead recorded the start of the command, which is where this rule places the violation, then we could ditch all of this code.

I think that might be possible, but something we could look at separately.

Reply:
Or maybe just record the range of the Command at least both the start and end somehow ...

Reply:
I changed it to record the beginning of the command instead. This indeed simplifies the code in the rule. However, at some point we need to perform the byte and position calculations, unfortunately, which is `CommandVisitor` now.

Going from an absolute byte position to a character doesn't seem to be obvious. That's what `characterPosition(of:)` is doing. In case you have a better idea, I'm open for it.

Reply:
And I've just changed it to a range. We also need to know where the command end, because this is the beginning of a new region.

```swift
// swiftlint:disable all
```

would otherwise not trigger `blanket_disable_command` anymore as it were already disabled at the position where it triggered. 🤯

Reply:
So I had this as a quick hack, storing `start` separately - next would have been to move it into a range - https://github.com/realm/SwiftLint/compare/main...mildm8nnered:mildm8nnered-invalid-swiftlint-command-rule?expand=1, so I think yours is fine.